syne_tune/optimizer/schedulers/searchers/multi_fidelity_searcher.py

```diff
@@ -0,0 +1,151 @@
+
+logger = logging.getLogger(__name__)
+
+
+class IndependentMultiFidelitySearcher(BaseSearcher):
+    """
+    Searcher for the multi-fidelity setting which fits independent models for each
+    resource level as as proposed by Falkner et al.
+
+    | BOHB: Robust and Efficient Hyperparameter Optimization at Scale
+    | S. Falkner and A. Klein and F. Hutter
+    | Proceedings of the 35th International Conference on Machine Learning
+
+    # TODO: Update docstrings
+
+    :param config_space: Configuration space
+    :param points_to_evaluate: List of configurations to be evaluated
+        initially (in that order). Each config in the list can be partially
+        specified, or even be an empty dict. For each hyperparameter not
+        specified, the default value is determined using a midpoint heuristic.
+        If ``None`` (default), this is mapped to ``[dict()]``, a single default config
+        determined by the midpoint heuristic. If ``[]`` (empty list), no initial
+        configurations are specified.
+    """
+
+    def __init__(
+        self,
+        config_space: Dict[str, Any],
+        searcher_cls: Optional[Union[str, SingleObjectiveBaseSearcher]] = "kde",
+        points_to_evaluate: Optional[List[dict]] = None,
+        random_seed: Optional[int] = None,
+        searcher_kwargs: dict[str, Any] = None,
+    ):
+        super().__init__(
+            config_space,
+            points_to_evaluate=points_to_evaluate,
+            random_seed=random_seed,
+        )
+
+        if searcher_kwargs is None:
+            self.searcher_kwargs = dict()
+        else:
+            self.searcher_kwargs = searcher_kwargs
+
+        if isinstance(searcher_cls, str):
+            assert searcher_cls in searcher_cls_dict
+            self.searcher_cls = searcher_cls_dict.get(searcher_cls)
+        else:
+            self.searcher_cls = searcher_cls
+        self.models = OrderedDict()
+
+    def initialize_model(self):
+        return self.searcher_cls(
+            config_space=self.config_space,
+            random_seed=self.random_seed,
+            **self.searcher_kwargs,
+        )
+
+    def suggest(self, **kwargs) -> Optional[Dict[str, Any]]:
+        """Suggest a new configuration.
+
+        Note: Query :meth:`_next_points_to_evaluate` for initial configs to return
+        first.
+
+        :param kwargs: Extra information may be passed from scheduler to
+            searcher
+        :return: New configuration. The searcher may return None if a new
+            configuration cannot be suggested. In this case, the tuning will
+            stop. This happens if searchers never suggest the same config more
+            than once, and all configs in the (finite) search space are
+            exhausted.
+        """
+        suggestion = self._next_points_to_evaluate()
+        if suggestion is not None:
+            return suggestion
+
+        if len(self.models) == 0:
+            return {
```

Comment:
```suggestion
        # in case we have not seen observations, we have no searcher and sample at random
        if len(self.models) == 0:
            return {
```